src/registrar/management/commands/transfer_transition_domains_to_domains.py

```diff
@@ -0,0 +1,323 @@
+
+            Created Domains:
+            {domains_to_create}
+
+            Updated Domains:
+            {updated_domain_entries}
+
+            {termColors.ENDC}
+            """,
+        )
+
+    def handle(
+        self,
+        **options,
+    ):
+        """Parse entries in TransitionDomain table
+        and create (or update) corresponding entries in the
+        Domain and DomainInvitation tables."""
+
+        # grab command line arguments and store locally...
+        debug_on = options.get("debug")
+        debug_max_entries_to_parse = int(
+            options.get("limitParse")
+        )  # set to 0 to parse all entries
+
+        self.print_debug_mode_statements(debug_on, debug_max_entries_to_parse)
+
+        # domains to ADD
+        domains_to_create = []
+        domain_invitations_to_create = []
+        # domains we UPDATED
+        updated_domain_entries = []
+        # domains we SKIPPED
+        skipped_domain_entries = []
+        # domainInvitations we SKIPPED
+        skipped_domain_invitations = []
+        # if we are limiting our parse (for testing purposes, keep
+        # track of total rows parsed)
+        total_rows_parsed = 0
+
+        logger.info(
+            f"""{termColors.OKGREEN}
+            ==========================
+            Beginning Data Transfer
+            ==========================
+            {termColors.ENDC}"""
+        )
+
+        for transition_domain in TransitionDomain.objects.all():
+            transition_domain_name = transition_domain.domain_name
+            transition_domain_status = transition_domain.status
+            transition_domain_email = transition_domain.username
+
+            # Check for existing domain entry
+            try:
+                # DEBUG:
+                self.print_debug(
+                    debug_on,
+                    f"""{termColors.OKCYAN}
+                    Processing Transition Domain: {transition_domain_name}, {transition_domain_status}, {transition_domain_email}
+                    {termColors.ENDC}""",  # noqa
+                )
+
+                # get the existing domain
+                target_domain = Domain.objects.get(name=transition_domain_name)
+
+                # DEBUG:
+                self.print_debug(
+                    debug_on,
+                    f"""{termColors.YELLOW}
+                    > Found existing domain entry for: {transition_domain_name}, {target_domain.state}
+                    {termColors.ENDC}""",  # noqa
+                )
+
+                # for existing entry, update the status to
+                # the transition domain status
+                update_made = self.update_domain_status(
+                    transition_domain, target_domain, debug_on
+                )
+                if update_made:
+                    updated_domain_entries.append(transition_domain.domain_name)
+
+            except Domain.DoesNotExist:
+                already_in_to_create = next(
```

Comment:
This is a bit hard to read I would add a comment here

Reply:
I'm going to change this to use .exists() in an if-else statement